modules/xorg/hm.nix

```
@@ -0,0 +1,34 @@
+{ pkgs, config, lib, ... }:
+
+with config.lib.stylix;
+let
+ slideshowScript = pkgs.writeScript "script.sh" ''
+   # This controls (in seconds) when to switch to the next image
+   INTERVAL=${builtins.toString config.stylix.wallpaper.delay}
+   imagearray=(${builtins.toString config.stylix.wallpaper.images})
+   
+   while true; do
+   	${pkgs.feh}/bin/feh --no-fehbg --bg-scale ''${imagearray[ $RANDOM % ''${#imagearray[@]} ]}
+   	sleep $INTERVAL
+   done
+ '';
+in {
+ options.stylix.targets.xorg.enable =
+   config.lib.stylix.mkEnableTarget
+     "the desktop background using Feh"
+     (with config.xsession.windowManager; bspwm.enable
+       || herbstluftwm.enable
+       || i3.enable
+       || spectrwm.enable
+       || xmonad.enable);
+
+ config.xsession.initExtra = lib.mkIf config.stylix.targets.xorg.enable 
+   (if (isAnimation config.stylix.wallpaper) then ''
+     ${pkgs.xwinwrap}/bin/xwinwrap -fs -ni -b -nf -ov -- mpv -wid WID --loop --no-audio ${config.stylix.wallpaper.animation}
+   '' else
+     (if (isVideo config.stylix.wallpaper) then ''
+       ${pkgs.xwinwrap}/bin/xwinwrap -fs -ni -b -nf -ov -- mpv -wid WID --loop --no-audio ${config.stylix.wallpaper.video}
+     '' else
```

Should we refactor into a variable:

```suggestion
 config.xsession.initExtra = let 
   command = value: ''
     ${pkgs.xwinwrap}/bin/xwinwrap -fs -ni -b -nf -ov -- mpv -wid WID --loop --no-audio ${value}
   ''; 
 in lib.mkIf config.stylix.targets.xorg.enable 
   (if (isAnimation config.stylix.wallpaper) then (command config.stylix.wallpaper.animation) else
     (if (isVideo config.stylix.wallpaper) then (command config.stylix.wallpaper.video) else
```